firefox-install-demo.toml

```diff
@@ -9,95 +9,131 @@ start_date = "2021-11-01"
 
 metrics = [
-    "install_success_rate",
-    "install_volume_by_os_version",
+    "install_volume_total",
+    "install_volume_win7",
+    "install_volume_win8",
+    "install_volume_win8_1",
+    "install_volume_win10"
 ]
 
 alerts = [
-    "install_success_rate",
+    "install_success_rate_win",
+    "install_success_rate_win8"
 ]
 
 [project.population]
 
 data_source = "firefox_installs"
-branches = [
-    "Win7",
-    "Win8",
-    "Win8.1",
-    "Win10"
-]
 monitor_entire_population = true
 
+[project.metric_groups.installs_by_os]
+friendly_name = "Installs by OS"
+description = "Breakdown of installs by OS"
+metrics = [
+    "install_volume_win7",
+    "install_volume_win8",
+    "install_volume_win8_1",
+    "install_volume_win10"
+]
+
+
 
 [metrics]
 
+[metrics.install_volume_total]
+data_source = "firefox_installs_all_versions"
+select_expression = "IF(os != 'other' AND LOGICAL_OR(succeeded), 1, 0) * 100"
+type = "scalar"
+
+[metrics.install_volume_total.statistics]
+sum = {}
+mean = {}
+
+[metrics.install_volume_win7]
+data_source = "firefox_installs"
+select_expression = "IF(os = 'Win7' AND LOGICAL_OR(succeeded), 1, 0) * 100"
+type = "scalar"
+
+[metrics.install_volume_win7.statistics]
+sum = {}
+mean = {}
+
+[metrics.install_volume_win8]
+data_source = "firefox_installs"
+select_expression = "IF(os = 'Win8' AND LOGICAL_OR(succeeded), 1, 0) * 100"
+type = "scalar"
+
+[metrics.install_volume_win8.statistics]
+sum = {}
+mean = {}
 
-[metrics.install_success_rate]
+[metrics.install_volume_win8_1]
 data_source = "firefox_installs"
-select_expression = "IF(LOGICAL_OR(succeeded), 1, 0) * 100"
+select_expression = "IF(os = 'Win8.1' AND LOGICAL_OR(succeeded), 1, 0) * 100"
 type = "scalar"
 
-[metrics.install_success_rate.statistics]
+[metrics.install_volume_win8_1.statistics]
+sum = {}
 mean = {}
 
-[metrics.install_volume_by_os_version]
+[metrics.install_volume_win8_1]
 data_source = "firefox_installs"
-select_expression = "IF(LOGICAL_OR(succeeded), 1, 0) * 100"
+select_expression = "IF(os = 'Win8.1' AND LOGICAL_OR(succeeded), 1, 0) * 100"
 type = "scalar"
 
-[metrics.install_volume_by_os_version.statistics]
+[metrics.install_volume_win8_1.statistics]
 sum = {}
+mean = {}
+
+[metrics.install_volume_win10]
+data_source = "firefox_installs"
+select_expression = "IF(os = 'Win10' AND LOGICAL_OR(succeeded), 1, 0) * 100"
+type = "scalar"
+
+[metrics.install_volume_win10.statistics]
+sum = {}
+mean = {}
 
 [alerts]
 
-[alerts.install_success_rate]
+[alerts.install_success_rate_win]
+type = "threshold"
+metrics = [
+    "install_volume_win7",
+    "install_volume_win8_1",
+    "install_volume_win10"
+]
+min = [90]
+
+[alerts.install_success_rate_win8]
 type = "threshold"
 metrics = [
-    "install_success_rate"
+    "install_volume_win8",
 ]
-# This one number is applied to each OS, unfortunately. Ideally
-# we would have per-branch configuration, since each Windows version
-# has its own fairly steady success rate.
 min = [86]
```

Comment:
Having separate metrics allows to define alerts with custom thresholds for each.